File: fontes/backend/java/fenix/src/main/java/com/t2ti/fenix/model/fiscal/FiscalNotaFiscalEntrada.java
/*******************************************************************************
Title: T2Ti ERP 3.0                                                                
Description: Model relacionado à tabela [FISCAL_NOTA_FISCAL_ENTRADA] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
package com.t2ti.fenix.model.fiscal;

import java.io.Serializable;
import java.math.BigDecimal;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.NamedQuery;
import javax.persistence.Table;

import com.t2ti.fenix.model.nfe.NfeCabecalho;

@Entity
@Table(name="FISCAL_NOTA_FISCAL_ENTRADA")
@NamedQuery(name="FiscalNotaFiscalEntrada.findAll", query="SELECT t FROM FiscalNotaFiscalEntrada t")
public class FiscalNotaFiscalEntrada implements Serializable {
	private static final long serialVersionUID = 1L;

    @Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;

    @Column(name="COMPETENCIA")
	private String competencia;

    @Column(name="CFOP_ENTRADA")
	private Integer cfopEntrada;

    @Column(name="VALOR_RATEIO_FRETE")
	private BigDecimal valorRateioFrete;

    @Column(name="VALOR_CUSTO_MEDIO")
	private BigDecimal valorCustoMedio;

    @Column(name="VALOR_ICMS_ANTECIPADO")
	private BigDecimal valorIcmsAntecipado;

    @Column(name="VALOR_BC_ICMS_ANTECIPADO")
	private BigDecimal valorBcIcmsAntecipado;

    @Column(name="VALOR_BC_ICMS_CREDITADO")
	private BigDecimal valorBcIcmsCreditado;

    @Column(name="VALOR_BC_PIS_CREDITADO")
	private BigDecimal valorBcPisCreditado;

    @Column(name="VALOR_BC_COFINS_CREDITADO")
	private BigDecimal valorBcCofinsCreditado;

    @Column(name="VALOR_BC_IPI_CREDITADO")
	private BigDecimal valorBcIpiCreditado;

    @Column(name="CST_CREDITO_ICMS")
	private String cstCreditoIcms;

    @Column(name="CST_CREDITO_PIS")
	private String cstCreditoPis;

    @Column(name="CST_CREDITO_COFINS")
	private String cstCreditoCofins;

    @Column(name="CST_CREDITO_IPI")
	private String cstCreditoIpi;

    @Column(name="VALOR_ICMS_CREDITADO")
	private BigDecimal valorIcmsCreditado;

    @Column(name="VALOR_PIS_CREDITADO")
	private BigDecimal valorPisCreditado;

    @Column(name="VALOR_COFINS_CREDITADO")
	private BigDecimal valorCofinsCreditado;

    @Column(name="VALOR_IPI_CREDITADO")
	private BigDecimal valorIpiCreditado;

    @Column(name="QTDE_PARCELA_CREDITO_PIS")
	private Integer qtdeParcelaCreditoPis;

    @Column(name="QTDE_PARCELA_CREDITO_COFINS")
	private Integer qtdeParcelaCreditoCofins;

    @Column(name="QTDE_PARCELA_CREDITO_ICMS")
	private Integer qtdeParcelaCreditoIcms;

    @Column(name="QTDE_PARCELA_CREDITO_IPI")
	private Integer qtdeParcelaCreditoIpi;

    @Column(name="ALIQUOTA_CREDITO_ICMS")
	private BigDecimal aliquotaCreditoIcms;

    @Column(name="ALIQUOTA_CREDITO_PIS")
	private BigDecimal aliquotaCreditoPis;

    @Column(name="ALIQUOTA_CREDITO_COFINS")
	private BigDecimal aliquotaCreditoCofins;

    @Column(name="ALIQUOTA_CREDITO_IPI")
	private BigDecimal aliquotaCreditoIpi;

    @ManyToOne
	@JoinColumn(name="ID_NFE_CABECALHO")
    private NfeCabecalho nfeCabecalho;

	public FiscalNotaFiscalEntrada() {
		//construtor padrão
	}

    public Integer getId() {
		return this.id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

    public String getCompetencia() {
		return this.competencia;
	}

	public void setCompetencia(String competencia) {
		this.competencia = competencia;
	}

    public Integer getCfopEntrada() {
		return this.cfopEntrada;
	}

	public void setCfopEntrada(Integer cfopEntrada) {
		this.cfopEntrada = cfopEntrada;
	}

    public BigDecimal getValorRateioFrete() {
		return this.valorRateioFrete;
	}

	public void setValorRateioFrete(BigDecimal valorRateioFrete) {
		this.valorRateioFrete = valorRateioFrete;
	}

    public BigDecimal getValorCustoMedio() {
		return this.valorCustoMedio;
	}

	public void setValorCustoMedio(BigDecimal valorCustoMedio) {
		this.valorCustoMedio = valorCustoMedio;
	}

    public BigDecimal getValorIcmsAntecipado() {
		return this.valorIcmsAntecipado;
	}

	public void setValorIcmsAntecipado(BigDecimal valorIcmsAntecipado) {
		this.valorIcmsAntecipado = valorIcmsAntecipado;
	}

    public BigDecimal getValorBcIcmsAntecipado() {
		return this.valorBcIcmsAntecipado;
	}

	public void setValorBcIcmsAntecipado(BigDecimal valorBcIcmsAntecipado) {
		this.valorBcIcmsAntecipado = valorBcIcmsAntecipado;
	}

    public BigDecimal getValorBcIcmsCreditado() {
		return this.valorBcIcmsCreditado;
	}

	public void setValorBcIcmsCreditado(BigDecimal valorBcIcmsCreditado) {
		this.valorBcIcmsCreditado = valorBcIcmsCreditado;
	}

    public BigDecimal getValorBcPisCreditado() {
		return this.valorBcPisCreditado;
	}

	public void setValorBcPisCreditado(BigDecimal valorBcPisCreditado) {
		this.valorBcPisCreditado = valorBcPisCreditado;
	}

    public BigDecimal getValorBcCofinsCreditado() {
		return this.valorBcCofinsCreditado;
	}

	public void setValorBcCofinsCreditado(BigDecimal valorBcCofinsCreditado) {
		this.valorBcCofinsCreditado = valorBcCofinsCreditado;
	}

    public BigDecimal getValorBcIpiCreditado() {
		return this.valorBcIpiCreditado;
	}

	public void setValorBcIpiCreditado(BigDecimal valorBcIpiCreditado) {
		this.valorBcIpiCreditado = valorBcIpiCreditado;
	}

    public String getCstCreditoIcms() {
		return this.cstCreditoIcms;
	}

	public void setCstCreditoIcms(String cstCreditoIcms) {
		this.cstCreditoIcms = cstCreditoIcms;
	}

    public String getCstCreditoPis() {
		return this.cstCreditoPis;
	}

	public void setCstCreditoPis(String cstCreditoPis) {
		this.cstCreditoPis = cstCreditoPis;
	}

    public String getCstCreditoCofins() {
		return this.cstCreditoCofins;
	}

	public void setCstCreditoCofins(String cstCreditoCofins) {
		this.cstCreditoCofins = cstCreditoCofins;
	}

    public String getCstCreditoIpi() {
		return this.cstCreditoIpi;
	}

	public void setCstCreditoIpi(String cstCreditoIpi) {
		this.cstCreditoIpi = cstCreditoIpi;
	}

    public BigDecimal getValorIcmsCreditado() {
		return this.valorIcmsCreditado;
	}

	public void setValorIcmsCreditado(BigDecimal valorIcmsCreditado) {
		this.valorIcmsCreditado = valorIcmsCreditado;
	}

    public BigDecimal getValorPisCreditado() {
		return this.valorPisCreditado;
	}

	public void setValorPisCreditado(BigDecimal valorPisCreditado) {
		this.valorPisCreditado = valorPisCreditado;
	}

    public BigDecimal getValorCofinsCreditado() {
		return this.valorCofinsCreditado;
	}

	public void setValorCofinsCreditado(BigDecimal valorCofinsCreditado) {
		this.valorCofinsCreditado = valorCofinsCreditado;
	}

    public BigDecimal getValorIpiCreditado() {
		return this.valorIpiCreditado;
	}

	public void setValorIpiCreditado(BigDecimal valorIpiCreditado) {
		this.valorIpiCreditado = valorIpiCreditado;
	}

    public Integer getQtdeParcelaCreditoPis() {
		return this.qtdeParcelaCreditoPis;
	}

	public void setQtdeParcelaCreditoPis(Integer qtdeParcelaCreditoPis) {
		this.qtdeParcelaCreditoPis = qtdeParcelaCreditoPis;
	}

    public Integer getQtdeParcelaCreditoCofins() {
		return this.qtdeParcelaCreditoCofins;
	}

	public void setQtdeParcelaCreditoCofins(Integer qtdeParcelaCreditoCofins) {
		this.qtdeParcelaCreditoCofins = qtdeParcelaCreditoCofins;
	}

    public Integer getQtdeParcelaCreditoIcms() {
		return this.qtdeParcelaCreditoIcms;
	}

	public void setQtdeParcelaCreditoIcms(Integer qtdeParcelaCreditoIcms) {
		this.qtdeParcelaCreditoIcms = qtdeParcelaCreditoIcms;
	}

    public Integer getQtdeParcelaCreditoIpi() {
		return this.qtdeParcelaCreditoIpi;
	}

	public void setQtdeParcelaCreditoIpi(Integer qtdeParcelaCreditoIpi) {
		this.qtdeParcelaCreditoIpi = qtdeParcelaCreditoIpi;
	}

    public BigDecimal getAliquotaCreditoIcms() {
		return this.aliquotaCreditoIcms;
	}

	public void setAliquotaCreditoIcms(BigDecimal aliquotaCreditoIcms) {
		this.aliquotaCreditoIcms = aliquotaCreditoIcms;
	}

    public BigDecimal getAliquotaCreditoPis() {
		return this.aliquotaCreditoPis;
	}

	public void setAliquotaCreditoPis(BigDecimal aliquotaCreditoPis) {
		this.aliquotaCreditoPis = aliquotaCreditoPis;
	}

    public BigDecimal getAliquotaCreditoCofins() {
		return this.aliquotaCreditoCofins;
	}

	public void setAliquotaCreditoCofins(BigDecimal aliquotaCreditoCofins) {
		this.aliquotaCreditoCofins = aliquotaCreditoCofins;
	}

    public BigDecimal getAliquotaCreditoIpi() {
		return this.aliquotaCreditoIpi;
	}

	public void setAliquotaCreditoIpi(BigDecimal aliquotaCreditoIpi) {
		this.aliquotaCreditoIpi = aliquotaCreditoIpi;
	}

    public NfeCabecalho getNfeCabecalho() {
		return this.nfeCabecalho;
	}

	public void setNfeCabecalho(NfeCabecalho nfeCabecalho) {
		this.nfeCabecalho = nfeCabecalho;
	}

		
}

File: fontes/backend/php/fenix/src/model/fiscal/FiscalNotaFiscalEntrada.php
<?php
/*******************************************************************************
Title: T2Ti ERP 3.0
Description: Model relacionado à tabela [FISCAL_NOTA_FISCAL_ENTRADA] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
declare(strict_types=1);

class FiscalNotaFiscalEntrada extends EloquentModel implements \JsonSerializable
{
    /**
     * The table associated with the model.
     *
     * @var string
     */
    protected $table = 'FISCAL_NOTA_FISCAL_ENTRADA';

    /**
     * Eager Loading - Relacionamentos que devem ser sempre carregados por padrão
     *
     * @var array
     */
    protected $with = ['nfeCabecalho', ];
	
    /**
     * Relations
     */
    public function nfeCabecalho()
    {
    	return $this->belongsTo(NfeCabecalho::class, 'ID_NFE_CABECALHO', 'ID');
    }


    /**
     * Gets e Sets
     */
    public function getIdAttribute() 
	{
		return $this->attributes['ID'];
	}

	public function setIdAttribute($id) 
	{
		$this->attributes['ID'] = $id;
	}

    public function getCompetenciaAttribute() 
	{
		return $this->attributes['COMPETENCIA'];
	}

	public function setCompetenciaAttribute($competencia) 
	{
		$this->attributes['COMPETENCIA'] = $competencia;
	}

    public function getCfopEntradaAttribute() 
	{
		return $this->attributes['CFOP_ENTRADA'];
	}

	public function setCfopEntradaAttribute($cfopEntrada) 
	{
		$this->attributes['CFOP_ENTRADA'] = $cfopEntrada;
	}

    public function getValorRateioFreteAttribute() 
	{
		return $this->attributes['VALOR_RATEIO_FRETE'];
	}

	public function setValorRateioFreteAttribute($valorRateioFrete) 
	{
		$this->attributes['VALOR_RATEIO_FRETE'] = $valorRateioFrete;
	}

    public function getValorCustoMedioAttribute() 
	{
		return $this->attributes['VALOR_CUSTO_MEDIO'];
	}

	public function setValorCustoMedioAttribute($valorCustoMedio) 
	{
		$this->attributes['VALOR_CUSTO_MEDIO'] = $valorCustoMedio;
	}

    public function getValorIcmsAntecipadoAttribute() 
	{
		return $this->attributes['VALOR_ICMS_ANTECIPADO'];
	}

	public function setValorIcmsAntecipadoAttribute($valorIcmsAntecipado) 
	{
		$this->attributes['VALOR_ICMS_ANTECIPADO'] = $valorIcmsAntecipado;
	}

    public function getValorBcIcmsAntecipadoAttribute() 
	{
		return $this->attributes['VALOR_BC_ICMS_ANTECIPADO'];
	}

	public function setValorBcIcmsAntecipadoAttribute($valorBcIcmsAntecipado) 
	{
		$this->attributes['VALOR_BC_ICMS_ANTECIPADO'] = $valorBcIcmsAntecipado;
	}

    public function getValorBcIcmsCreditadoAttribute() 
	{
		return $this->attributes['VALOR_BC_ICMS_CREDITADO'];
	}

	public function setValorBcIcmsCreditadoAttribute($valorBcIcmsCreditado) 
	{
		$this->attributes['VALOR_BC_ICMS_CREDITADO'] = $valorBcIcmsCreditado;
	}

    public function getValorBcPisCreditadoAttribute() 
	{
		return $this->attributes['VALOR_BC_PIS_CREDITADO'];
	}

	public function setValorBcPisCreditadoAttribute($valorBcPisCreditado) 
	{
		$this->attributes['VALOR_BC_PIS_CREDITADO'] = $valorBcPisCreditado;
	}

    public function getValorBcCofinsCreditadoAttribute() 
	{
		return $this->attributes['VALOR_BC_COFINS_CREDITADO'];
	}

	public function setValorBcCofinsCreditadoAttribute($valorBcCofinsCreditado) 
	{
		$this->attributes['VALOR_BC_COFINS_CREDITADO'] = $valorBcCofinsCreditado;
	}

    public function getValorBcIpiCreditadoAttribute() 
	{
		return $this->attributes['VALOR_BC_IPI_CREDITADO'];
	}

	public function setValorBcIpiCreditadoAttribute($valorBcIpiCreditado) 
	{
		$this->attributes['VALOR_BC_IPI_CREDITADO'] = $valorBcIpiCreditado;
	}

    public function getCstCreditoIcmsAttribute() 
	{
		return $this->attributes['CST_CREDITO_ICMS'];
	}

	public function setCstCreditoIcmsAttribute($cstCreditoIcms) 
	{
		$this->attributes['CST_CREDITO_ICMS'] = $cstCreditoIcms;
	}

    public function getCstCreditoPisAttribute() 
	{
		return $this->attributes['CST_CREDITO_PIS'];
	}

	public function setCstCreditoPisAttribute($cstCreditoPis) 
	{
		$this->attributes['CST_CREDITO_PIS'] = $cstCreditoPis;
	}

    public function getCstCreditoCofinsAttribute() 
	{
		return $this->attributes['CST_CREDITO_COFINS'];
	}

	public function setCstCreditoCofinsAttribute($cstCreditoCofins) 
	{
		$this->attributes['CST_CREDITO_COFINS'] = $cstCreditoCofins;
	}

    public function getCstCreditoIpiAttribute() 
	{
		return $this->attributes['CST_CREDITO_IPI'];
	}

	public function setCstCreditoIpiAttribute($cstCreditoIpi) 
	{
		$this->attributes['CST_CREDITO_IPI'] = $cstCreditoIpi;
	}

    public function getValorIcmsCreditadoAttribute() 
	{
		return $this->attributes['VALOR_ICMS_CREDITADO'];
	}

	public function setValorIcmsCreditadoAttribute($valorIcmsCreditado) 
	{
		$this->attributes['VALOR_ICMS_CREDITADO'] = $valorIcmsCreditado;
	}

    public function getValorPisCreditadoAttribute() 
	{
		return $this->attributes['VALOR_PIS_CREDITADO'];
	}

	public function setValorPisCreditadoAttribute($valorPisCreditado) 
	{
		$this->attributes['VALOR_PIS_CREDITADO'] = $valorPisCreditado;
	}

    public function getValorCofinsCreditadoAttribute() 
	{
		return $this->attributes['VALOR_COFINS_CREDITADO'];
	}

	public function setValorCofinsCreditadoAttribute($valorCofinsCreditado) 
	{
		$this->attributes['VALOR_COFINS_CREDITADO'] = $valorCofinsCreditado;
	}

    public function getValorIpiCreditadoAttribute() 
	{
		return $this->attributes['VALOR_IPI_CREDITADO'];
	}

	public function setValorIpiCreditadoAttribute($valorIpiCreditado) 
	{
		$this->attributes['VALOR_IPI_CREDITADO'] = $valorIpiCreditado;
	}

    public function getQtdeParcelaCreditoPisAttribute() 
	{
		return $this->attributes['QTDE_PARCELA_CREDITO_PIS'];
	}

	public function setQtdeParcelaCreditoPisAttribute($qtdeParcelaCreditoPis) 
	{
		$this->attributes['QTDE_PARCELA_CREDITO_PIS'] = $qtdeParcelaCreditoPis;
	}

    public function getQtdeParcelaCreditoCofinsAttribute() 
	{
		return $this->attributes['QTDE_PARCELA_CREDITO_COFINS'];
	}

	public function setQtdeParcelaCreditoCofinsAttribute($qtdeParcelaCreditoCofins) 
	{
		$this->attributes['QTDE_PARCELA_CREDITO_COFINS'] = $qtdeParcelaCreditoCofins;
	}

    public function getQtdeParcelaCreditoIcmsAttribute() 
	{
		return $this->attributes['QTDE_PARCELA_CREDITO_ICMS'];
	}

	public function setQtdeParcelaCreditoIcmsAttribute($qtdeParcelaCreditoIcms) 
	{
		$this->attributes['QTDE_PARCELA_CREDITO_ICMS'] = $qtdeParcelaCreditoIcms;
	}

    public function getQtdeParcelaCreditoIpiAttribute() 
	{
		return $this->attributes['QTDE_PARCELA_CREDITO_IPI'];
	}

	public function setQtdeParcelaCreditoIpiAttribute($qtdeParcelaCreditoIpi) 
	{
		$this->attributes['QTDE_PARCELA_CREDITO_IPI'] = $qtdeParcelaCreditoIpi;
	}

    public function getAliquotaCreditoIcmsAttribute() 
	{
		return $this->attributes['ALIQUOTA_CREDITO_ICMS'];
	}

	public function setAliquotaCreditoIcmsAttribute($aliquotaCreditoIcms) 
	{
		$this->attributes['ALIQUOTA_CREDITO_ICMS'] = $aliquotaCreditoIcms;
	}

    public function getAliquotaCreditoPisAttribute() 
	{
		return $this->attributes['ALIQUOTA_CREDITO_PIS'];
	}

	public function setAliquotaCreditoPisAttribute($aliquotaCreditoPis) 
	{
		$this->attributes['ALIQUOTA_CREDITO_PIS'] = $aliquotaCreditoPis;
	}

    public function getAliquotaCreditoCofinsAttribute() 
	{
		return $this->attributes['ALIQUOTA_CREDITO_COFINS'];
	}

	public function setAliquotaCreditoCofinsAttribute($aliquotaCreditoCofins) 
	{
		$this->attributes['ALIQUOTA_CREDITO_COFINS'] = $aliquotaCreditoCofins;
	}

    public function getAliquotaCreditoIpiAttribute() 
	{
		return $this->attributes['ALIQUOTA_CREDITO_IPI'];
	}

	public function setAliquotaCreditoIpiAttribute($aliquotaCreditoIpi) 
	{
		$this->attributes['ALIQUOTA_CREDITO_IPI'] = $aliquotaCreditoIpi;
	}


    /**
     * Mapping
     */
    public function mapear($objeto)
    {
        if (isset($objeto)) {
            isset($objeto->id) ? $this->setIdAttribute($objeto->id) : $this->setIdAttribute(null);

			$this->setCompetenciaAttribute($objeto->competencia);
			$this->setCfopEntradaAttribute($objeto->cfopEntrada);
			$this->setValorRateioFreteAttribute($objeto->valorRateioFrete);
			$this->setValorCustoMedioAttribute($objeto->valorCustoMedio);
			$this->setValorIcmsAntecipadoAttribute($objeto->valorIcmsAntecipado);
			$this->setValorBcIcmsAntecipadoAttribute($objeto->valorBcIcmsAntecipado);
			$this->setValorBcIcmsCreditadoAttribute($objeto->valorBcIcmsCreditado);
			$this->setValorBcPisCreditadoAttribute($objeto->valorBcPisCreditado);
			$this->setValorBcCofinsCreditadoAttribute($objeto->valorBcCofinsCreditado);
			$this->setValorBcIpiCreditadoAttribute($objeto->valorBcIpiCreditado);
			$this->setCstCreditoIcmsAttribute($objeto->cstCreditoIcms);
			$this->setCstCreditoPisAttribute($objeto->cstCreditoPis);
			$this->setCstCreditoCofinsAttribute($objeto->cstCreditoCofins);
			$this->setCstCreditoIpiAttribute($objeto->cstCreditoIpi);
			$this->setValorIcmsCreditadoAttribute($objeto->valorIcmsCreditado);
			$this->setValorPisCreditadoAttribute($objeto->valorPisCreditado);
			$this->setValorCofinsCreditadoAttribute($objeto->valorCofinsCreditado);
			$this->setValorIpiCreditadoAttribute($objeto->valorIpiCreditado);
			$this->setQtdeParcelaCreditoPisAttribute($objeto->qtdeParcelaCreditoPis);
			$this->setQtdeParcelaCreditoCofinsAttribute($objeto->qtdeParcelaCreditoCofins);
			$this->setQtdeParcelaCreditoIcmsAttribute($objeto->qtdeParcelaCreditoIcms);
			$this->setQtdeParcelaCreditoIpiAttribute($objeto->qtdeParcelaCreditoIpi);
			$this->setAliquotaCreditoIcmsAttribute($objeto->aliquotaCreditoIcms);
			$this->setAliquotaCreditoPisAttribute($objeto->aliquotaCreditoPis);
			$this->setAliquotaCreditoCofinsAttribute($objeto->aliquotaCreditoCofins);
			$this->setAliquotaCreditoIpiAttribute($objeto->aliquotaCreditoIpi);

			// vincular objetos
			$nfeCabecalho = new NfeCabecalho();
			$nfeCabecalho->mapear($objeto->nfeCabecalho);
			$this->nfeCabecalho()->associate($nfeCabecalho);

        }
    }

    /**
     * Serialization
     * {@inheritdoc}
     */
    public function jsonSerialize()
    {
        return [
			'id' => $this->getIdAttribute(),
			'competencia' => $this->getCompetenciaAttribute(),
			'cfopEntrada' => $this->getCfopEntradaAttribute(),
			'valorRateioFrete' => $this->getValorRateioFreteAttribute(),
			'valorCustoMedio' => $this->getValorCustoMedioAttribute(),
			'valorIcmsAntecipado' => $this->getValorIcmsAntecipadoAttribute(),
			'valorBcIcmsAntecipado' => $this->getValorBcIcmsAntecipadoAttribute(),
			'valorBcIcmsCreditado' => $this->getValorBcIcmsCreditadoAttribute(),
			'valorBcPisCreditado' => $this->getValorBcPisCreditadoAttribute(),
			'valorBcCofinsCreditado' => $this->getValorBcCofinsCreditadoAttribute(),
			'valorBcIpiCreditado' => $this->getValorBcIpiCreditadoAttribute(),
			'cstCreditoIcms' => $this->getCstCreditoIcmsAttribute(),
			'cstCreditoPis' => $this->getCstCreditoPisAttribute(),
			'cstCreditoCofins' => $this->getCstCreditoCofinsAttribute(),
			'cstCreditoIpi' => $this->getCstCreditoIpiAttribute(),
			'valorIcmsCreditado' => $this->getValorIcmsCreditadoAttribute(),
			'valorPisCreditado' => $this->getValorPisCreditadoAttribute(),
			'valorCofinsCreditado' => $this->getValorCofinsCreditadoAttribute(),
			'valorIpiCreditado' => $this->getValorIpiCreditadoAttribute(),
			'qtdeParcelaCreditoPis' => $this->getQtdeParcelaCreditoPisAttribute(),
			'qtdeParcelaCreditoCofins' => $this->getQtdeParcelaCreditoCofinsAttribute(),
			'qtdeParcelaCreditoIcms' => $this->getQtdeParcelaCreditoIcmsAttribute(),
			'qtdeParcelaCreditoIpi' => $this->getQtdeParcelaCreditoIpiAttribute(),
			'aliquotaCreditoIcms' => $this->getAliquotaCreditoIcmsAttribute(),
			'aliquotaCreditoPis' => $this->getAliquotaCreditoPisAttribute(),
			'aliquotaCreditoCofins' => $this->getAliquotaCreditoCofinsAttribute(),
			'aliquotaCreditoIpi' => $this->getAliquotaCreditoIpiAttribute(),
			'nfeCabecalho' => $this->nfeCabecalho,
        ];
    }
}